src/Shared/CommunicationsUtilities.cs

```diff
@@ -665,12 +665,78 @@ internal static int GetIntegerVariableOrDefault(string environmentVariable, int
             return defaultValue;
         }
 
+        /// <summary>
+        /// Writes trace information to a log file
+        /// </summary>
+        internal static void Trace<T>(string format, T arg0)
```

Comment:
I want to raise awareness that there are costs for generic methods in this context (extra JIT, possibly multiple JITted methods for each T held in memory, etc).

It's probably not that big of a deal, but can have costs that are harder to pin down.

Any reason for not using an `object` parameter instead (i.e. many primitives that would box passed as the arg?

Reply:
My understanding is that avoiding boxing is an explicit goal of this PR. I think I'd also favor generics over boxing in the no-tracing case.

Reply:
The main motivator was that these methods are called even when the tracing is disabled. I wanted to prevent allocation of ```object[]``` and boxing of value types. These overloads are short, and when T is value type, it's usually ```int```. I'm not sure that CLR is JITing this method for every T. I believe there is one compiled method for any reference type and specialized variations just for value types.